esmvaltool/diag_scripts/ocean/diagnostic_timeseries.py

```diff
@@ -62,11 +63,65 @@ def timeplot(cube, **kwargs):
 
     Needed because iris version 1.13 fails due to the time axis.
     """
-    if iris.__version__ > '2.0':
-        qplt.plot(cube, kwargs)
-    else:
-        times = diagtools.timecoord_to_float(cube.coord('time'))
-        plt.plot(times, cube.data, **kwargs)
+    cubedata = np.ma.array(cube.data)
+    if len(cubedata.compressed()) == 1:
+        plt.axhline(cubedata.compressed(), **kwargs)
+        return
+
+    times = diagtools.cube_time_to_float(cube)
+    plt.plot(times, cubedata, **kwargs)
+
+
+def moving_average(cube, window):
+    """
+    Make a moving average plot.
+
+    the window is a string which isa number and a measuremet of time.
+    """
+    window = window.split()
+    window_len = int(window[0]) / 2.
+    window_units = str(window[1])
+
+    if window_units not in ['days', 'day', 'dy',
+                            'months', 'month', 'mn',
+                            'years', 'yrs', 'year', 'yr']:
+        raise ValueError("Window_units not recognised: %s",
+                         str(window_units))
+
+    data = cube.data
+    time_coord = cube.coord('time')
+    times = time_coord.units.num2date(time_coord.points)
+
+    datetime = diagtools.guess_calendar_datetime(cube)
+
+    output = []
+    times = [datetime(time.year, time.month, time.day, time.hour, time.minute)
+             for time in times]
+    times = np.array(times)
+
+    for time in times:
+        if window_units in ['years', 'yrs', 'year', 'yr']:
+            tmin = datetime(time.year - window_len, time.month, time.day,
+                            time.hour, time.minute)
+            tmax = datetime(time.year + window_len, time.month, time.day,
+                            time.hour, time.minute)
+
+        if window_units in ['months', 'month', 'mn']:
+            tmin = datetime(time.year, time.month - window_len, time.day,
+                            time.hour, time.minute)
+            tmax = datetime(time.year, time.month + window_len, time.day,
+                            time.hour, time.minute)
```

Comment:
can you rename your variable `time` to something that's not an actual Python module? This can bring problems if we actually will import time in a script talking to this one

Reply:
Replaced with `time_itr`.